matrix_client/client.py

```diff
@@ -106,7 +117,7 @@ def login_with_password(self, username, password, limit=1):
         Args:
             username (str): Account username
             password (str): Account password
-            limit (int): How many messages to return when syncing.
+            limit (int): Deprecated. How many messages to return when syncing.
```

Comment:
As much as I'd prefer to just chop it off entirely, I can just do something like 

``self.sync_filter = '{ "room": { "timeline" : "limit" : ' + limit + ' } }'``

and pledge to remove it on next release in favor of a real api?